### PR TITLE
Add Drupal 7 integration

### PR DESCRIPTION
**Pull request by t2t2**
Drupal 7 integration 

Generates a trace that looks like this (collapsed spans just contain more PDO.prepare & pdostatement.execute spans):

image

Most of the new files are from drupal 7.82 source, these are in their own commit (f1331d1) for ignoring purposes.

### Readiness checklist
- [x] (only for Members) Changelog has been added to the release document.
- [x] Tests added for this feature/bug.

### CODE REVIEW

File: src/DDTrace/Integrations/Drupal/DrupalIntegration.php [hunk 1]
```diff
@@ -51,6 +53,78 @@ public function init()
...
+                $rootSpan->overwriteOperationName($path);
+            }
+        });
+
+        // Can't directly trace functions called by set_error_handler & set_exception_handler
+        \DDTrace\trace_function('_drupal_error_handler_real', function (SpanData $span, $args) {
+            $span->name = '_drupal_error_handler';
+            $span->meta[Tag::COMPONENT] = 'drupal';
+            $span->meta[Tag::ERROR_MSG] = $args[1];
+            $span->meta[TAG::ERROR_TYPE] = 'error handler';
```
Comment (flands) [inline src/DDTrace/Integrations/Drupal/DrupalIntegration.php:109]: Any reason why this is `TAG` versus `Tag`?

Comment (t2t2) [reply src/DDTrace/Integrations/Drupal/DrupalIntegration.php]: 🤷  fixed

**Review by mhennoch — APPROVED** [review]

**Review by seemk — APPROVED** [review]

**Comment by breedx-splk** [comment]
Most of those failed tests can be fixed up by incorporating the `xfail_tests` from https://github.com/DataDog/dd-trace-php/pull/1275/files. Recommend trying that please.

**Comment by t2t2** [comment]
Synced other versions too with current xfail files in upstream (mostly for changes in https://github.com/DataDog/dd-trace-php/commit/2f3367824ac508e8174c96251d32854041dc1c87 and https://github.com/DataDog/dd-trace-php/commit/a3be1a26b89c3251ded4f2102695865d5508c383 )

**Pull request merged**